### DISCUSSION TITLE
Auto fix fieldalignment with this tool at the end of generation steps

### DISCUSSION THREAD

**Discussion by frederikhors** [99designs/gqlgen#1895] [Ideas]
### What happened?

In generated code we have a lot of structs with misaligned fields (the issue is well described here: https://qvault.io/golang/struct-field-ordering-memory/).

### What did you expect?

To auto-fix this we can launch at the end of generation this tool:

`go install golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@latest`

with this command:

```
fieldalignment -fix <package_path>
```

I don't know how to create a PR for this otherwise I would have already done it.

Thanks for your precious work!

**Comment by switchupcb** [comment]
Here is a start @frederikhors: https://github.com/99designs/gqlgen/pull/2069
